sui_core/src/authority_server.rs

```diff
@@ -58,331 +77,319 @@ impl AuthorityServer {
-                or signature / committee verification operation fails. The client
-                should know better than give invalid input. All conditions can be
-                trivially checked on the client side, so there should be no surprises
-                here for well behaved clients.
-            */
-            if let Err(err) = one_chunk {
-                // If the response channel is closed there is no much we can do
-                // to handle the error result.
-                let _ = channel.sink().send(serialize_error(&err).into()).await;
-                return;
-            }
-            let mut one_chunk = one_chunk.unwrap();
-
-            // Process each message
-            while let Some((_message, _buffer)) = one_chunk.pop_front() {
-                if let Some(reply) = self.handle_one_message(_message, &mut channel).await {
-                    let status = channel.sink().send(reply.into()).await;
-                    if let Err(error) = status {
-                        error!("Failed to send query response: {error}");
+            .map_err(|e| tonic::Status::internal(e.to_string()))?;
+
+        let payload = BincodeEncodedPayload::try_from(&response)
+            .map_err(|e| tonic::Status::internal(e.to_string()))?;
+
+        Ok(tonic::Response::new(payload))
+    }
+
+    async fn object_info(
+        &self,
+        request: tonic::Request<BincodeEncodedPayload>,
+    ) -> Result<tonic::Response<BincodeEncodedPayload>, tonic::Status> {
+        let request: ObjectInfoRequest = request
+            .into_inner()
+            .deserialize()
+            .map_err(|e| tonic::Status::invalid_argument(e.to_string()))?;
+
+        let response = self
+            .state
+            .handle_object_info_request(request)
+            .await
+            .map_err(|e| tonic::Status::internal(e.to_string()))?;
+
+        let payload = BincodeEncodedPayload::try_from(&response)
+            .map_err(|e| tonic::Status::internal(e.to_string()))?;
+
+        Ok(tonic::Response::new(payload))
+    }
+
+    async fn transaction_info(
+        &self,
+        request: tonic::Request<BincodeEncodedPayload>,
+    ) -> Result<tonic::Response<BincodeEncodedPayload>, tonic::Status> {
+        let request: TransactionInfoRequest = request
+            .into_inner()
+            .deserialize()
+            .map_err(|e| tonic::Status::invalid_argument(e.to_string()))?;
+
+        let response = self
+            .state
+            .handle_transaction_info_request(request)
+            .await
+            .map_err(|e| tonic::Status::internal(e.to_string()))?;
+
+        let payload = BincodeEncodedPayload::try_from(&response)
+            .map_err(|e| tonic::Status::internal(e.to_string()))?;
+
+        Ok(tonic::Response::new(payload))
+    }
+
+    type BatchInfoStream = BoxStream<'static, Result<BincodeEncodedPayload, tonic::Status>>;
+
+    async fn batch_info(
```

Comment:
In PR https://github.com/MystenLabs/sui/pull/1670 this is now in large part done in the authority.rs that returns a stream ready to feed out to the client.

Reply:
So are you saying that this implementation will get simpiler?